mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -738,6 +797,36 @@ impl MakerSwap {
         };
         swap_events.push(MakerSwapEvent::MakerPaymentInstructionsReceived(instructions));
 
+        let taker_amount = MmNumber::from(self.taker_amount.clone());
+        let remote_version = self
+            .r()
+            .data
+            .taker_version
+            .ok_or("No swap protocol version".to_owned())?;
+        let is_burn_active = SwapFeature::is_active(SwapFeature::SendToPreBurnAccount, remote_version);
+        let dex_fee = dex_fee_from_taker_coin(
+            self.taker_coin.deref(),
+            &self.r().data.maker_coin,
+            &taker_amount,
+            Some(self.r().other_taker_coin_htlc_pub.to_vec().as_ref()),
+            Some(is_burn_active),
+        );
```

Comment:
i don't think this would be dev-friendly to upgrade to the next feature set. I like the versioning mechanism and think we should have one, but also think it should be abstracted away in each entities impl (here DexFee).

so how i imagine something like this would be is: taker and maker settle on a version together (deterministically, pick the min version for e.g. or have a deterministic mapping rules), and we pass that version to whoever needs it to apply some version specific rules (in here this is the DexFee, so something like `DexFee::using_version(version, taker_coin, maker_coin, etc...)` and inside we match against the version and apply the appropriate rule),

Reply:
I started to do this but realised there is a problem here:
We support both legacy and TPU protocols, each of both may have own versioning. So it looks like it's better to deal with versions outside the lower level components like DexFee (which is called both from legacy and TPU), don't you think?